### PR TITLE
bots: Refactor youtube bot.

### PR DESCRIPTION
**Pull request by SivagiriVisakan**
I have modified the Youtube bot (as a [GCI task](https://codein.withgoogle.com/dashboard/task-instances/5262318340407296/) ) to use the Youtube Data API instead of BeautifulSoup.
The bot now responds with single video as well as a list of videos.
There are a couple of things left to do , but I have opened this PR for feedback on any changes that maybe  needed.

TO-DO:

- [x] Handle errors from API
- [x] Add more tests
- [x] Add documentation

### CODE REVIEW

**Comment by showell** [comment]
I haven't done a detailed review yet, but this seems like a reasonable approach so far!

File: zulip_bots/zulip_bots/bots/youtube/youtube.py [hunk 1]
```diff
@@ -24,6 +25,14 @@ def usage(self):
 
     def initialize(self, bot_handler):
         self.config_info = bot_handler.get_config_info('youtube')
+        # Check if API key is valid. If it is not valid, don't run the bot.
+        try:
+            search_youtube('test', self.config_info['key'])
+        except HTTPError as e:
+            if (e.response.json()['error']['errors'][0]['reason'] == 'keyInvalid'):
```
Comment (SivagiriVisakan) [inline zulip_bots/zulip_bots/bots/youtube/youtube.py]: Not sure if this is a right approach to  check the API key.  Seeking advice as to how to proceed or if this approach is fine

File: zulip_bots/zulip_bots/bots/youtube/test_youtube.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                self.mock_http_conversation('test_single'):
+                    self.verify_reply('funny cats', bot_response)
+
+    def test_invalid_key(self) -> None:
+        bot = get_bot_message_handler(self.bot_name)
+        bot_handler = StubBotHandler()
+
+        with self.mock_config_info({'key': 'somethinginvalid', 'number_of_results': '5', 'video_region': 'US'}), \
+                self.mock_http_conversation('test_invalid_key'), \
+                self.assertRaises(SystemExit) as se:  # type: ignore
```
Comment (SivagiriVisakan) [inline zulip_bots/zulip_bots/bots/youtube/test_youtube.py]: Ignoring type here as it was causing mypy to fail with message : 
````
tools/run-mypy youtube
Running mypy for `zulip`.
There are no files to run mypy on.
Running mypy for `zulip_bots`.
zulip_bots/zulip_bots/bots/youtube/test_youtube.py:32: error: No overload variant of "assertRaises" of "TestCase" matches argument types [def (*args: Any) -> builtins.SystemExit]
```

**Comment by showell** [comment]
Merged, thanks @SivagiriVisakan!  I can tell a lot of careful work went into this!

**Pull request merged**